src/v3/src/util/getEventContext.ts

```diff
@@ -14,15 +14,44 @@ import { IdxTransaction } from '@okta/okta-auth-js';
 
 import { EventContext } from '../../../types';
 import { getV1ClassName } from '../../../v2/ion/ViewClassNamesFactory';
+import { IDX_STEP } from '../constants';
 import { getAuthenticatorKey } from './getAuthenticatorKey';
 import { isPasswordRecovery } from './isPasswordRecovery';
 
+export const getFormNameForTransaction = (transaction: IdxTransaction): string | undefined => {
+  const {
+    nextStep: { name: formName } = {},
+    neededToProceed,
+    context,
+    rawIdxState,
+  } = transaction;
+  const hasSkipRemediationOnly = neededToProceed?.length === 1 && neededToProceed[0].name === 'skip';
+  if (!neededToProceed?.length || hasSkipRemediationOnly) {
+    // no remediation or only skip remediation with success
+    if (context.success) {
+      return context.success.name;
+    }
+    // no remediation or only skip remediation with messages
+    if (context.messages) {
+      return 'terminal';
+    }
+    // no remediation or only skip remediation with messages for device enrollment state
+    // and the state is meant to be terminal state with different UI than the regular terminal view
+    // @ts-ignore deviceEnrollment missing from type
+    if (rawIdxState.deviceEnrollment) {
+      return IDX_STEP.DEVICE_ENROLLMENT_TERMINAL;
+    }
+  }
+  return formName;
+};
```

Comment:
The spec tested for proper formName values in the afterRender hook, I did some digging and found that we were missing some scenarios for applying the proper formName.